heapster-saw/src/Verifier/SAW/Heapster/Permissions.hs

```diff
@@ -4944,6 +4944,14 @@ buildDistPerms :: (forall a. Name a -> ValuePerm a) -> RAssign Name ps ->
 buildDistPerms _ MNil = DistPermsNil
 buildDistPerms f (ns :>: n) = DistPermsCons (buildDistPerms f ns) n (f n)
 
+-- | Like 'buildDistPerms', but accumulates some state through a @foldr@
+foldrAndBuildDistPerms :: (forall a. Name a -> b -> (ValuePerm a, b)) -> b ->
+                          RAssign Name ps -> (DistPerms ps, b)
+foldrAndBuildDistPerms _ b0 MNil = (DistPermsNil, b0)
+foldrAndBuildDistPerms f b0 (ns :>: n)
+  | (ps, b) <- foldrAndBuildDistPerms f b0 ns, (p, b') <- f n b
+  = (DistPermsCons ps n p, b')
+
```

Comment:
```suggestion
```
This function is not used anywhere, and is just a special case of `rlMapMWithAccum`.